internal/worker/api/server_test.go

```diff
@@ -0,0 +1,559 @@
+package api
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/renatoaguimaraes/job-scheduler/internal/worker/proto"
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker/conf"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+
+var config = conf.Config{ServerAddress: "localhost:8080", LogFolder: os.TempDir()}
+
+func TestStartAuthnAuthzAdminUser(t *testing.T) {
+	// load server credentials
+	servercred, err := loadServerCredentials(clientca, servercert, serverkey)
+	assert.Nil(t, err)
+	// creates server
+	serv, lis, err := CreateServer(config, servercred)
+	assert.Nil(t, err)
+	defer serv.Stop()
+	// starts the server
+	go func() {
+		serv.Serve(lis)
+	}()
+	// waits server bind
+	time.Sleep(time.Second)
```

Comment:
is this necessary?
the listener should already be accepting connections after `CreateServer` returns

Reply:
Right, doesn't need any more.